packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -757,7 +759,14 @@ export class AnchorSet implements Listenable<AnchorSetRootEvents>, AnchorLocator
 						continue;
 					}
 					emittedEvents?.push(event);
-					node.events.emit(event, node);
+					if (event === "childrenChangedAfterBatch") {
+						const fieldKeys = this.bufferedEvents
+							.filter((e) => e.node === node && e.event === event)
+							.map((e) => e.changedField as FieldKey);
```

Comment:
this should use brand for a safer conversion

Reply:
actually, this is not casting a string to FIeldKey, but casting away undefined. If unchecked, this should use `!` rather than `as` for much better clarity and type safety. I'm going to make it checked via ?? instead.

Reply:
Done.